flowapi/flowapi/user_model.py

```diff
@@ -76,16 +77,44 @@ def has_access(
             raise UserClaimsVerificationError("Claims verification failed.")
         return True
 
-    def can_run(self, *, query_kind: str, aggregation_unit: str) -> bool:
+    def _get_query_kind_and_aggregation_unit(self, query_json: dict) -> Tuple[str, str]:
         """
-        Returns true if the user can run this kind of query at this unit of aggregation.
+        Extract the query kind and aggregation unit from a query spec dict.
 
         Parameters
         ----------
-        query_kind : str
-            Kind of the query
-        aggregation_unit : str
-            Aggregation unit/level of resolution
+        query_json : dict
+            Dictionary containing a query spec
+
+        Returns
+        -------
+        str, str
+            Query kind and aggregation unit
+
+        """
+
+        try:
+            query_kind = query_json["query_kind"]
+            if query_kind == "spatial_aggregate":
+                return self._get_query_kind_and_aggregation_unit(
```

Comment:
Hmm, here we are extracting the aggregation unit from the input locations query. Is this really what we want in all cases? It works for e.g. daily/modal locations, but shouldn't it be possible to do a spatial aggregation on something that was previously non-aggregated? Or, say, to run a further aggregation to admin2-level of daily locations that are on admin3 level?

I don't consider this a blocker for this PR but I think I'd be much happier if we added a separate `aggregation_unit` parameter to the `spatial_aggregate` query itself, and only did some consistency checks with aggregation unit of the input query.

Reply:
Well at this time, `SpatialAggregate` doesn't support that, so it would seem a bit moot really. I think should we add support for a different approach to spatial aggregation, we'd want to revisit and see if we could find a better solution.